clients/roscpp/src/libros/transport/transport_tcp.cpp

```diff
@@ -42,6 +42,7 @@
 #include <boost/bind.hpp>
 #include <fcntl.h>
 #include <errno.h>
+#include <sys/socket.h>  // explicit include required for FreeBSD
```

Comment:
@dirk-thomas sorry for the late on this, but it seems to including `sys/socket.h` directly without any conditional guards would break the Windows build....